app/views/shared/_taxon.html.erb

```diff
@@ -1,30 +1,39 @@
 <%-
   taxon ||= @taxon
   include_article ||= false
+  one_name ||= false
   singular ||= false
+  exclude_rank ||= false
   no_common ||= false
   place ||= nil
   comname ||= common_taxon_name(taxon, place: place || @site.try(:place)).try(:name) unless no_common
+  truncate ||= false
+  trunclength ||= truncate ? 23 : 500
   if taxon.species_or_lower?
     pieces = taxon.name.split
     sciname = if taxon.variety? && pieces.size == 3
       "<span class=\"sciname #{taxon.rank}\">#{pieces[0..1].join(' ')} <span class=\"rank\">var.</span> #{pieces[2..-1].join(' ')}</span>"
     elsif taxon.subspecies? && pieces.size == 3
       "<span class=\"sciname #{taxon.rank}\">#{pieces[0..1].join(' ')} <span class=\"rank\">ssp.</span> #{pieces[2..-1].join(' ')}</span>"
     else
-      "<span class=\"sciname\">#{taxon.name}</span>"
+      "<span class=\"sciname\">#{truncate(taxon.name,length: trunclength)}</span>"
```

Comment:
if this just for display, CSS can do string truncation on the fly based on element size